cmd/api-mock/main.go

```diff
@@ -105,6 +115,159 @@ func NewUpgradePolicyStateEndpoint() *UpgradePolicyStateEndpoint {
 	}
 }
 
+type addonStatusStore struct {
+	data    map[addonStatusKey]addonStatus
+	dataMux sync.RWMutex
+}
+
+type addonStatusKey struct {
+	clusterID string
+	addonID   string
+}
+
+type addonStatus struct {
+	AddonID       string `json:"addon_id"`
+	CorrelationID string `json:"correlation_id"`
+	// We dont care about this unmarshalling this field.
+	StatusConditions []interface{} `json:"status_conditions"`
```

Comment:
```suggestion
```
I don't notice this field being used here, so no point worth making the json decoder go through unmarshalling/marshalling this field as well. 
wdyt

Reply:
I left it there to document the existence of that attribute. Since we are mocking the api, it'll be good to have all the attributes present in the API.

Reply:
Gotcha, makes sense.